src/routes/github/github-oauth-router.ts

```diff
@@ -250,6 +264,25 @@ const appendJiraHostIfNeeded = (url: string, jiraHost: string): string => {
 	return url;
 };
 
+const renewGitHubToken = async (githubRefreshToken: string, gitHubAppConfig: GitHubAppConfig, jiraHost: string, logger: Logger) => {
+	try {
+		const clientSecret = await getCloudOrGHESAppClientSecret(gitHubAppConfig, jiraHost);
+		if (clientSecret) {
+			const gitHubAnonymousClient = await createAnonymousClientByGitHubAppId(gitHubAppConfig?.gitHubAppId, jiraHost, logger);
+			const { accessToken, refreshToken } = await gitHubAnonymousClient.renewGitHubToken({
+				refreshToken: githubRefreshToken,
+				clientId: gitHubAppConfig.clientId,
+				clientSecret
+			});
+			return { accessToken, refreshToken };
```

Comment:
kinda redundent to deconstruct, then reconstruct the object?

Reply:
Refactored.